### PR TITLE
Tutorials docs

### PR DESCRIPTION
**Pull request by brianignacio5**
Adding tutorial documentation for the extension. 

- [x] Extension Setup
- [x] SDK Configuration editor, Build, Flash, monitor tutorial with blink_example
- [x] Debugging with blink_example
- [x] Code coverage
- [x] Partition table editor
- [x] NVS Partition editor
- [x] Heap (System view) tracing
- [x] CMakeLists.txt editor
- [x] Rainmaker integration
- [x] eFuse Summary

### CODE REVIEW

File: docs/COVERAGE.md [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+
 ## Requirements
 
-Your ESP-IDF project should be configured to generate gcda/gcno coverage files using gcov. Please take a look at the [ESP-IDF gcov example](https://github.com/espressif/esp-idf/tree/master/examples/system/gcov) to see how to set up your project.
+Your ESP-IDF project should be configured to generate gcda/gcno coverage files using gcov as shown in [GCOV Code coverage](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/app_trace.html#gcov-source-code-coverage). Please take a look at the [ESP-IDF gcov example](https://github.com/espressif/esp-idf/tree/master/examples/system/gcov) as example project.
 
-This extension also requires `gcovr` to generate JSON and HTML reports from generated files. This is installed as part of this extension ESP-IDF Debug Adapter Python requirements when following the **ESP-IDF: Configure ESP-IDF extension** command.
+This extension requires `gcovr` to generate JSON and HTML reports from generated files. This is installed as part of this extension ESP-IDF Debug Adapter Python requirements when following the **ESP-IDF: Configure ESP-IDF extension** command.
```
[the remark this answers is not in the document]
Comment (brianignacio5) [reply docs/COVERAGE.md:11]: It is old information that needs to be updated, was removed from debug adapter and moved to extension requirements.txt. Will make updates as suggested !

File: docs/SETUP.md [hunk 2]
```diff
@@ -91,7 +110,12 @@ where EXTENSION_PATH is
 - `%USERPROFILE%\.vscode\extensions\espressif.esp-idf-extension-VERSION` on Windows
 - `$HOME/.vscode/extensions/espressif.esp-idf-extension-VERSION` on Linux/MacOS.
 
-## Example configuration setting values
+# UI Manual Configuration
+
+This is the same as [JSON Manual Configuration](#JSON-Manual-Configuration) but the name of each configuration setting is the description given in the [ESP-IDF Settings](./SETTINGS.md).
+You also need to install extension and debug adapter python packages requirements.txt.
```
[the remark this answers is not in the document]
Comment (brianignacio5) [reply docs/SETUP.md:116]: The thing is, if the user wants to manually set the extension, both JSON and UI Manual configuration will not install extension and debug adapter requirements.txt, so I think they need to be reminded to install them in their python virtual environment. Not sure about where will be good to add this text though.

**Review by krzychb — CHANGES_REQUESTED** [review]
@brianignacio5 please see some comments.

File: docs/tutorial/debugging.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+13. If you press “Step Out (Shift + F11)” instead, then debugger will step outside the subroutine call.
+
+<p align="center">
+  <img src="../../media/tutorials/debug/step_out.png" alt="Step out">
+</p>
+
+In this particular case debugger stepped inside `vTaskDelay(CONFIG_BLINK_PERIOD / portTICK_PERIOD_MS)` and effectively moved to `tasks.c` code. See [Why stepping with “next” does not bypass subroutine calls?](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/jtag-debugging/tips-and-quirks.html#jtag-debugging-tip-why-next-works-as-step) for potential limitations using the `next` command.
+
+# Watching and setting program variables
```
[the remark this answers is not in the document]
Comment (brianignacio5) [reply docs/tutorial/debugging.md:94]: Currently is not part of vscode debug features. Reference : [vscode issue](https://github.com/microsoft/vscode/issues/105406)

File: docs/tutorial/debugging.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+14. Stop debugging by pressing "Stop (Shift + F5)". Build and flash the code to the ESP and restart the debugger by pressing F5. Once the application is halted, set a breakpoint in the line where `i++` is.
+
+15. Next in the `Watch` sub-window on the `Run` icon in the Activity Bar on the side of Visual Studio Code, click the + and enter `i` to start watching its value.
+
+16. Continue the program execution by pressing F5. Each time the program is halted, you will see `i` being incremented.
+
+<p align="center">
+  <img src="../../media/tutorials/debug/watch_set_program_vars.png" alt="Watch program variables">
+</p>
+
```
[the remark this answers is not in the document]
Comment (brianignacio5) [reply docs/tutorial/debugging.md:107]: not implemented yet in the debug adapter.

[the remark this answers is not in the document]
Comment (brianignacio5) [reply docs/tutorial/debugging.md:111]: Yes, I had add it in the debug tutorial, I wasn't sure at first but did tested and it works.

**Comment by brianignacio5** [comment]
@krzychb @gerekon Can we close this ? Or do you think there is something we could improve/add ?

**Comment by krzychb** [comment]
> @krzychb @gerekon Can we close this ? Or do you think there is something we could improve/add ?

LGTM

**Pull request merged**